### PR TITLE
Strip HTML when receiving HTML server errors in cloud API

### PR DESCRIPTION
**Pull request by suricactus**
It breaks the QField message log, as it supports HTML.

| before | after | 
|-|-|
| image | image |

### CODE REVIEW

**Comment by duke-nyuki** [comment]
Task linked: [QF-3981 QField shows HTML when plain HTML error pages are received](https://app.clickup.com/t/2192114/QF-3981)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 84a16911ffff5fb678546ddf0295d98250f2e50e: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-5115-0112674b8c97fa055d1c7656ae93e05fe62b8652-android-arm64.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-5115-0112674b8c97fa055d1c7656ae93e05fe62b8652-android-arm-neon.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-5115-0112674b8c97fa055d1c7656ae93e05fe62b8652-android-x64.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-5115-0112674b8c97fa055d1c7656ae93e05fe62b8652-android-x86.apk)
- [**all access arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-5115-0112674b8c97fa055d1c7656ae93e05fe62b8652-android-arm64.apk)
- [all access arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-5115-0112674b8c97fa055d1c7656ae93e05fe62b8652-android-arm-neon.apk)
- [all access x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-5115-0112674b8c97fa055d1c7656ae93e05fe62b8652-android-x64.apk)
- [all access x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-5115-0112674b8c97fa055d1c7656ae93e05fe62b8652-android-x86.apk)
</details>

File: src/core/qfieldcloudconnection.cpp [hunk 1]
```diff
@@ -507,29 +507,31 @@ QFieldCloudConnection::CloudError::CloudError( QNetworkReply *reply )
...
   else
   {
-    QgsMessageLog::logMessage( QStringLiteral( "%1\n%2\n%3\n%4" ).arg( errorMessage, httpErrorMessage, mPayload ).arg( reply->errorString() ) );
+    QgsMessageLog::logMessage( QStringLiteral( "%1\n%2\n%3\n%4" ).arg( errorMessage, httpErrorMessage, payloadPlainText ).arg( reply->errorString() ) );
   }
 
   // strip HTML tags
-  QTextDocument doc;
-  doc.setHtml( errorMessage );
-  errorMessage = doc.toPlainText();
```
Comment (nirvn) [inline src/core/qfieldcloudconnection.cpp:532]: Why was this not working before?

Comment (suricactus) [reply src/core/qfieldcloudconnection.cpp:532]: This specific one was working, I just find the `QTextDocumentFragment` syntax more elegant.

However, we send the message to `QgsMessageLog` a few lines above, therefore we see the funky message log.

Comment (nirvn) [reply src/core/qfieldcloudconnection.cpp:532]: @suricactus , ah got it.

**Review by nirvn — APPROVED** [review]

**Pull request merged**